libstorage/drivers/storage/cinder/storage/cinder_storage.go

```diff
@@ -330,7 +333,7 @@ func translateVolume(
 			libstorageAttachment := &types.VolumeAttachment{
 				VolumeID:   attachment.VolumeID,
 				InstanceID: &types.InstanceID{ID: attachment.ServerID, Driver: cinder.Name},
-				DeviceName: attachment.Device,
+				DeviceName: resolveDeviceName(attachment.VolumeID, attachment.Device),
```

Comment:
This instead needs to match the volume ID to those found in the LocalDevices map, which right now is empty, and pull the device name from that. This code may be running on a different node than where the volume is attached.

Each driver has to do this a bit differently, but here is an example from [EBS](https://github.com/rexray/rexray/blob/d14c3e3073be0b0955648b7de8a0a882a87977d7/libstorage/drivers/storage/ebs/storage/ebs_storage.go#L953-L971) and [ScaleIO](https://github.com/rexray/rexray/blob/d14c3e3073be0b0955648b7de8a0a882a87977d7/libstorage/drivers/storage/scaleio/storage/scaleio_storage.go#L214-L230)